src/main/java/com/purbon/kafka/topology/exceptions/ConfigurationException.java

```diff
@@ -1,8 +1,6 @@
 package com.purbon.kafka.topology.exceptions;
 
-import java.io.IOException;
-
-public class ConfigurationException extends IOException {
+public class ConfigurationException extends RuntimeException {
```

Comment:
can you please elaborate a bit on this change go RuntimeException?

Reply:
The previously checked exception was never handled explicitly but just rethrown, so I changed it to an unchecked one (and I wanted to use the  ConfigurationException when a URL value could not be parsed without adding a throws clause in the getter in Configuration#getKSQLServer)

Reply:
I wonder, if instead of making it runtime, we could keep it as an exception and properly handle it. Not sure about your experiences, but I kinda prefer always explicit exception handling as much as possible. What do you think?

Reply:
I think in the end the question is: Can we properly handle or solve the issue in a catch block when the user provided an invalid configuration?

There is a little section for checked vs unchecked here:
https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html